functional/tools/stresstest/stresstest.go

```diff
@@ -0,0 +1,152 @@
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"strconv"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/spiffe/spire/proto/api/registration"
+	"github.com/spiffe/spire/proto/common"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+const (
+	serverAddr           = "localhost:8081"
+	parentSpiffeIDPrefix = "spiffe://example.org/spire/agent/join_token/"
+	spiffeIDPrefix       = "spiffe://example.org/"
+	workloadPath         = "/go/src/github.com/spiffe/spire/functional/tools/workload/workload"
+)
+
+type workloadStats struct {
+	runtime time.Duration
+	output  string
+	success bool
+}
+
+func main() {
+	token := flag.String("token", "", "Join token used in server and agent")
+	users := flag.Int("workloads", 5, "Number of workloads to run in parallel")
+	timeout := flag.Int("timeout", 15, "Total time to run test")
+	ttl := flag.Int("ttl", 120, "SVID TTL")
+	flag.Parse()
+
+	if *token == "" {
+		flag.Usage()
+		return
+	}
+
+	c, err := newRegistrationClient(serverAddr)
+	if err != nil {
+		panic(err)
+	}
+
+	// Create users and register workloads
+	for i := 0; i < *users; i++ {
+		uid := 1000 + i
+
+		fmt.Printf("Creating user %d\n", uid)
+
+		// Create user
+		o, err := exec.Command("bash", "-c", fmt.Sprintf("useradd --uid %d user%d", uid, uid)).CombinedOutput() //adduser --uid 1111 --disabled-password --shell /bin/bash --ingroup spire spire
```

Comment:
Perhaps we can do some safety check to ensure this is being run inside our special container? I'd hate to accidentally run this on my Linux laptop :)

Reply:
And by accidentally, I mean on purpose but without knowing the tool will create new system users